platforms/software/ivy/src/integTest/groovy/org/gradle/api/publish/ivy/IvyPublishJavaIntegTest.groovy

```diff
@@ -958,6 +958,46 @@ class IvyPublishJavaIntegTest extends AbstractIvyPublishIntegTest {
         }
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/26163")
+    def "can publish java-library with lazy capabilities"() {
```

Comment:
In addition to testing this through publishing, we should also verify that the capabilities are correct when tested through a `ResolutionResult`.

Also I find it odd that we have this under an "ivy" test. This really doesn't test any ivy behavior. Do we have tests more specific to GMM?

Reply:
I only found tests related to Ivy and Maven specifically. This Ivy one seemed like the simplest one, I couldn't find an equivalent Maven test.

Reply:
I spoke with sterling about the testing here. He mentioned the possibility of dogfooding this feature by replacing the existing `ProjectDerivedCapability`. Do you think that is feasible here?

I still think we should have a `ResolutionResult` test in addition to a publishing one. 

In terms of maven, there is a similar `MavenPublishJavaIntegTest`, though in reality I feel that this feature is not java-related at all. We should be able to test this pretty easily without applying the java-library plugin at all.

Reply:
I pushed up a commit with a new test and fixed a bug when using `requireCapability` with a provider. I think we can probably hold off on the dogfooding for another PR.